hack/ci/check.sh

```diff
@@ -18,9 +18,86 @@
 
 set -ex
 
-./kubectl-direct_csi info
-./kubectl-direct_csi drives list
-./kubectl-direct_csi drives list | grep -q Available
-./kubectl-direct_csi drives format --all
-sleep 5
-./kubectl-direct_csi drives list
+args=("$@")
+
+function setup_lvm() {
+    sudo truncate --size=1G disk-{1..4}.img
+    for disk in disk-{1..4}.img; do sudo losetup --find $disk; done
+    devices=( $(for disk in disk-{1..4}.img; do sudo losetup --noheadings --output NAME --associated $disk; done) )
+    sudo pvcreate "${devices[@]}"
+    vgname="test-vg-$RANDOM"
+    sudo vgcreate "$vgname" "${devices[@]}"
+    for i in {1..4}; do sudo lvcreate --size=800MiB "$vgname"; done
+}
+
+function install_directcsi()
+{
+    ./kubectl-direct_csi install --image "direct-csi:${args[0]}"
+    sleep 1m
+    kubectl describe pods -n direct-csi-min-io
+    ./kubectl-direct_csi info
+    ./kubectl-direct_csi drives list | grep -q Available
+    ./kubectl-direct_csi drives format --all
+    sleep 5
+    ./kubectl-direct_csi drives list --all
+}
+
+function deploy_minio()
+{
+    kubectl apply -f hack/ci/minio.yaml
+    sleep 1m
+    kubectl get pods
+
+    runningpods=$(kubectl get pods --field-selector=status.phase=Running --no-headers | wc -l)
+    if [[ $runningpods -ne 4 ]]
+    then
+        echo "MinIO deployment failed"
+        exit 1
+    fi
+}
+
+function uninstall_minio()
+{
+    kubectl delete -f hack/ci/minio.yaml
+    sleep 1m
+    kubectl delete pvc --all
```

Comment:
If I understand correctly, this command stages deletion. You could remove above `sleep`